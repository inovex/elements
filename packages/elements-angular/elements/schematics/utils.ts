import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
