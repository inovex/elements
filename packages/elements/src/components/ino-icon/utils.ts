import { getAssetPath } from '@stencil/core';

let CACHED_MAP: Map<string, string>;

export function getIconMap(): Map<string, string> {
	if (!CACHED_MAP) {
		const win = window as any;
		win.inoIcons = win.inoIcons || {};
		CACHED_MAP = win.inoIcons.map = win.inoIcons.map || new Map();
	}
	return CACHED_MAP;
}

export function addIcons(icons: { [name: string]: string }) {
	const map = getIconMap();
	Object.keys(icons).forEach((name) => map.set(name, icons[name]));
}

export const getSrc = (src: string | undefined) => {
	if (isStr(src)) {
		src = src.trim();
		if (isSrc(src)) {
			return src;
		}
	}
	return null;
};

export function getName(name: string | undefined) {
	if (name && !isSrc(name)) {
		return name;
	}
	return undefined;
}

export function getUrl(src: string | undefined, icon: string | undefined) {
	let url = getSrc(src);

	if (url) {
		return url;
	}
	url = getName(icon);
	if (url) {
		return getNamedUrl(url);
	}

	url = getSrc(icon);
	if (url) {
		return url;
	}

	return null;
}

export function getNamedUrl(name: string) {
	const url = getIconMap().get(name);
	if (url) {
		return url;
	}
	return getAssetPath(`./ino-icon/${name}.svg`);
}

export const isSrc = (str: string) => str.length > 0 && /(\/|\.)/.test(str);

export const isStr = (val: any): val is string => typeof val === 'string';

export const toLower = (val: string) => val.toLowerCase();
