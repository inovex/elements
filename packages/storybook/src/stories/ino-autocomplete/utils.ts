import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
