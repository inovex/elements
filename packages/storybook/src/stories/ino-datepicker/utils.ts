import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
