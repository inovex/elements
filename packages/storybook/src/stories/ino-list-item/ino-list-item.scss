.story-ino-list-item {

  ino-list ino-icon {
    --ino-icon-height: 20px;
    --ino-icon-width: 20px;
  }
}
