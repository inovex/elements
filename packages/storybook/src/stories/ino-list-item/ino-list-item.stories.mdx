import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-list-item/readme.md";

<Meta title="Structure/ino-list-item/Documentation" />

<Description markdown={readme} />
