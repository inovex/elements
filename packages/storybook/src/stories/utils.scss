/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
