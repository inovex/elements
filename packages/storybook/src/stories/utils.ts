import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
