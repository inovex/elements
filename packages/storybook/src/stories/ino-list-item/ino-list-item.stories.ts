import { html } from 'lit-html';
import { defaultDecorator } from '../utils';
import { useEffect } from '@storybook/client-api';
import './ino-list-item.scss';

export default {
  title: 'Structure/ino-list-item',
  component: 'ino-list-item',
  decorators: [
    story => defaultDecorator(story, 'story-ino-list-item'),
    story => {
      useEffect(() => {
        const eventHandler = e => {
          const el = e.target;
          el.setAttribute('checked', e.detail);
          if (el.getAttribute('indeterminate') === 'true') {
            el.setAttribute('indeterminate', 'false');
          }
        };
        const selections = document.querySelectorAll('ino-checkbox, ino-radio');
        selections.forEach(s => s.addEventListener('checkedChange', eventHandler));
        return () => selections.forEach(s => s.removeEventListener('checkedChange', eventHandler));
      });
      return story();
    }
  ],
};

export const Playground = (args) => html`
  <ino-list two-lines="${args.twoLines}">
    <ino-list-item
      secondary-text="${args.secondaryText}"
      selected="${args.selected}"
      activated="${args.activated}"
      disabled="${args.disabled}"
      text="${args.text}">
    </ino-list-item>
  </ino-list>
`;
Playground.args = {
  twoLines: true,
  secondaryText: 'Second line text',
  selected: false,
  activated: false,
  disabled: false,
  text: 'List item',
}
Playground.argTypes = {
  twoLines: {
    description: 'Two Lines option of the parent list element'
  }
}

const exampleImg = html`
  <ino-img
    slot="leading"
    src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
    ratio-width="1"
    ratio-height="1"
  ></ino-img>
`;

export const States = () => html`
  <ino-list>
    <ino-list-item text="Simple item"></ino-list-item>
    <ino-list-item selected text="Selected item"></ino-list-item>
    <ino-list-item selected text="Selected item 2"></ino-list-item>
    <ino-list-item text="Simple item 2"></ino-list-item>
    <ino-list-item activated text="Activated item"></ino-list-item>
    <ino-list-item disabled text="Disabled item"></ino-list-item>
  </ino-list>
`;

export const GraphicAndMeta = () => html`
  <ino-list>
    <ino-list-item text="Lorem ipsum dolor sit">
      ${exampleImg}
      <p slot="trailing">$10.00</p>
    </ino-list-item>
    <ino-list-item text="Lorem ipsum dolor sit">
      ${exampleImg}
      <p slot="trailing">$10.00</p>
    </ino-list-item>
  </ino-list>
`;

export const Checkbox = () => html`
  <h4>Leading checkbox</h4>
  <ino-list>
    <ino-list-item text="First text item">
      <ino-checkbox slot="leading" selection></ino-checkbox>
    </ino-list-item>
    <ino-list-item text="Second text item">
      <ino-checkbox slot="leading" selection></ino-checkbox>
    </ino-list-item>
  </ino-list>

  <h4>Trailing checkbox</h4>
  <ino-list>
    <ino-list-item text="Lorem ipsum dolor sit">
      <ino-checkbox slot="trailing" selection></ino-checkbox>
    </ino-list-item>
    <ino-list-item text="Lorem ipsum dolor sit">
      <ino-checkbox slot="trailing" selection></ino-checkbox>
    </ino-list-item>
  </ino-list>

  <h4>Leading checkbox and trailing metadata</h4>
  <ino-list>
    <ino-list-item text="Lorem ipsum dolor sit">
      <ino-checkbox slot="leading" selection></ino-checkbox>
      <p slot="trailing">Some Metdata</p>
    </ino-list-item>
    <ino-list-item text="Lorem ipsum dolor sit">
      <ino-checkbox slot="leading" selection></ino-checkbox>
      <p slot="trailing">Some Metdata</p>
    </ino-list-item>
  </ino-list>
`;

export const Radio = () => html`
  <h4>Leading radio button</h4>
  <ino-list>
    <ino-list-item text="First text item">
      <ino-radio slot="leading" name="group-1"></ino-radio>
    </ino-list-item>
    <ino-list-item text="Second text item">
      <ino-radio slot="leading" name="group-1"></ino-radio>
    </ino-list-item>
  </ino-list>

  <h4>Trailing radio button</h4>
  <ino-list>
    <ino-list-item text="Lorem ipsum dolor sit">
      <ino-radio slot="trailing" name="group-5"></ino-radio>
    </ino-list-item>
    <ino-list-item text="Lorem ipsum dolor sit">
      <ino-radio slot="trailing" name="group-5"></ino-radio>
    </ino-list-item>
  </ino-list>

  <h4>Leading radio button and trailing metadata</h4>
  <ino-list>
    <ino-list-item text="Lorem ipsum dolor sit">
    <ino-radio slot="leading" name="group-6"></ino-radio>
      <p slot="trailing">Some Metadata</p>
    </ino-list-item>
    <ino-list-item text="Lorem ipsum dolor sit">
    <ino-radio slot="leading" name="group-6"></ino-radio>
      <p slot="trailing">Some Metadata</p>
    </ino-list-item>
  </ino-list>
`;

export const Slots = () => html`
  <h4>Primary Slot</h4>
  <ino-list>
    <ino-list-item>
      ${exampleImg}
      <a slot="primary" href="">Link Element</a>
    </ino-list-item>
  </ino-list>

  <h4>Primary and Secondary Slot</h4>
  <ino-list two-lines>
    <ino-list-item>
      ${exampleImg}
      <a slot="primary" href="">Primary Link Element</a>
      <a slot="secondary" href="">Secondary Link Element</a>
      <p slot="trailing">$10.00</p>
    </ino-list-item>
  </ino-list>
`;

export const TwoLines = () => html`
  <ino-list two-lines avatar>
    <ino-list-item text="Lorem ipsum dolor sit" secondary-text="Secondary Lorem ipsum dolor sit">
      ${exampleImg}
      <ino-icon slot="trailing" icon="info"></ino-icon>
    </ino-list-item>
    <ino-list-item text="Lorem ipsum dolor sit" secondary-text="Secondary Lorem ipsum dolor sit">
      ${exampleImg}
      <ino-icon slot="trailing" icon="info"></ino-icon>
    </ino-list-item>
  </ino-list>
`;
