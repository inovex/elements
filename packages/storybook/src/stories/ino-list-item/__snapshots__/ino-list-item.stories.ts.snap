// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-list-item Activated smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-list-item">
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-list-item secondary-text
                       selected="false"
                       activated="true"
                       disabled="false"
                       text="List item"
                       class="hydrated"
        >
          <li class="mdc-deprecated-list-item mdc-deprecated-list-item--activated mdc-ripple-upgraded"
              tabindex="-1"
          >
            <span class="mdc-deprecated-list-item__ripple">
            </span>
            <span class="mdc-deprecated-list-item__text">
              List item
            </span>
          </li>
        </ino-list-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-list-item Disabled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-list-item">
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-list-item secondary-text
                       selected="false"
                       activated="false"
                       disabled="true"
                       text="List item"
                       class="hydrated"
        >
          <li class="mdc-deprecated-list-item mdc-deprecated-list-item--disabled mdc-ripple-upgraded"
              tabindex="-1"
          >
            <span class="mdc-deprecated-list-item__ripple">
            </span>
            <span class="mdc-deprecated-list-item__text">
              List item
            </span>
          </li>
        </ino-list-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-list-item Graphics smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-list-item">
    <ino-list class="hydrated">
      <ul class="mdc-deprecated-list">
        <ino-list-item text="Lorem ipsum dolor sit"
                       class="hydrated"
        >
          <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
              tabindex="-1"
          >
            <span class="mdc-deprecated-list-item__ripple">
            </span>
            <span class="mdc-deprecated-list-item__graphic"
                  role="presentation"
            >
              <ino-img slot="leading"
                       src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
                       ratio-width="1"
                       ratio-height="1"
                       class="hydrated"
              >
                <div class="ino-img__composer"
                     style="padding-top: 100%;"
                >
                  <img class="ino-img__image"
                       src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
                  >
                  <div class="ino-img__fallback-icon">
                    <ino-icon class="hydrated ino-icon--color-scheme-primary">
                    </ino-icon>
                  </div>
                </div>
              </ino-img>
            </span>
            <span class="mdc-deprecated-list-item__text">
              Lorem ipsum dolor sit
            </span>
          </li>
        </ino-list-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-list-item MetaData smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-list-item">
    <ino-list class="hydrated">
      <ul class="mdc-deprecated-list">
        <ino-list-item text="Lorem ipsum dolor sit"
                       class="hydrated"
        >
          <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
              tabindex="-1"
          >
            <span class="mdc-deprecated-list-item__ripple">
            </span>
            <span class="mdc-deprecated-list-item__text">
              Lorem ipsum dolor sit
            </span>
            <span class="mdc-deprecated-list-item__meta">
              <p slot="trailing">
                $10.00
              </p>
            </span>
          </li>
        </ino-list-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-list-item Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-list-item">
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-list-item secondary-text
                       selected="false"
                       activated="false"
                       disabled="false"
                       text="List item"
                       class="hydrated"
        >
          <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
              tabindex="-1"
          >
            <span class="mdc-deprecated-list-item__ripple">
            </span>
            <span class="mdc-deprecated-list-item__text">
              List item
            </span>
          </li>
        </ino-list-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-list-item SecondaryText smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-list-item">
    <ino-list two-lines="true"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list mdc-deprecated-list--two-line">
        <ino-list-item secondary-text="Second line text"
                       selected="false"
                       activated="false"
                       disabled="false"
                       text="List item"
                       class="hydrated"
        >
          <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
              tabindex="-1"
          >
            <span class="mdc-deprecated-list-item__ripple">
            </span>
            <span class="mdc-deprecated-list-item__text">
              <span class="mdc-deprecated-list-item__primary-text">
                List item
              </span>
              <span class="mdc-deprecated-list-item__secondary-text">
                Second line text
              </span>
            </span>
          </li>
        </ino-list-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-list-item Selected smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-list-item">
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-list-item secondary-text
                       selected="true"
                       activated="false"
                       disabled="false"
                       text="List item"
                       class="hydrated"
        >
          <li class="mdc-deprecated-list-item mdc-deprecated-list-item--selected mdc-ripple-upgraded"
              tabindex="-1"
          >
            <span class="mdc-deprecated-list-item__ripple">
            </span>
            <span class="mdc-deprecated-list-item__text">
              List item
            </span>
          </li>
        </ino-list-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-list-item Text smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-list-item">
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-list-item secondary-text
                       selected="false"
                       activated="false"
                       disabled="false"
                       text="simple-item"
                       class="hydrated"
        >
          <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
              tabindex="-1"
          >
            <span class="mdc-deprecated-list-item__ripple">
            </span>
            <span class="mdc-deprecated-list-item__text">
              simple-item
            </span>
          </li>
        </ino-list-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-list-item WithCheckbox smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-list-item">
    <ino-list class="hydrated">
      <ul class="mdc-deprecated-list">
        <ino-list-item text="Lorem ipsum dolor sit"
                       class="hydrated"
        >
          <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
              tabindex="-1"
          >
            <span class="mdc-deprecated-list-item__ripple">
            </span>
            <span class="mdc-deprecated-list-item__graphic"
                  role="presentation"
            >
              <ino-checkbox slot="leading"
                            class="hydrated"
              >
                <input type="hidden"
                       class="aux-input"
                       name="undefined"
                       value
                >
              </ino-checkbox>
            </span>
            <span class="mdc-deprecated-list-item__text">
              Lorem ipsum dolor sit
            </span>
          </li>
        </ino-list-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-list-item WithRadio smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-list-item">
    <ino-list class="hydrated">
      <ul class="mdc-deprecated-list">
        <ino-list-item text="Lorem ipsum dolor sit"
                       class="hydrated"
        >
          <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
              tabindex="-1"
          >
            <span class="mdc-deprecated-list-item__ripple">
            </span>
            <span class="mdc-deprecated-list-item__graphic"
                  role="presentation"
            >
              <ino-radio slot="leading"
                         selection
                         class="hydrated"
              >
                <input type="hidden"
                       class="aux-input"
                       name="undefined"
                       value
                >
              </ino-radio>
            </span>
            <span class="mdc-deprecated-list-item__text">
              Lorem ipsum dolor sit
            </span>
          </li>
        </ino-list-item>
      </ul>
    </ino-list>
  </div>
</div>
`;
